### ISSUE TITLE
Extend WORKERFS to accept urls?

### ISSUE THREAD

**Issue by jvail** [emscripten-core/emscripten#15034]
I wonder what you think about extending the WORKERFS as such that it would also accept urls (not only blobs and files).

I found a few old issues (like #2047) that are pretty close to what I am looking for: Mount a url into WORKEFS to be able to use it like a blob/file. Since XHR is available in workers it could synchronously fetch the binary data from the url.

My use case is that I am porting a GIS library (an sqlite extension) and a dependency (proj) requires quite a large (6mb) database to perform any kind of geometry transformation. So instead of fetching the entire database and mount, copy it into the WORKERFS why not just do mount('https://github.com/jvail/spl.js/blob/main/dist/proj/proj.db', 'proj')?

Basically something like this but without a VFS in sqlite: https://github.com/psanford/sqlite3vfshttp

The other alternative would be to implement an sqlite virtual filesystem but the first solution seems to be much easier and - I guess - might be a useful feature within emscripten. What do you think?
